core/src/main/java/com/sequenceiq/cloudbreak/reactor/handler/cluster/ClusterStartHandler.java

```diff
@@ -95,25 +95,37 @@ public void accept(Event<ClusterStartRequest> event) {
             int requestId;
-            // Update the RDC of data lake and update the HMS configuration
-            if (stack.getType() == StackType.WORKLOAD && datalakeStack.isPresent() &&
-                    isDatalakeCreatedAfterDataHub(datalakeStack.get(), stack) &&
-                    entitlementService.isDatalakeLightToMediumMigrationEnabled(ThreadBasedUserCrnProvider.getAccountId())) {
+            // Re-configuring DH using the Remote Data Context of Data lake.
+            boolean clusterDataHub = stack.getType().equals(StackType.WORKLOAD);
+            boolean dlIsRebuild = datalakeStack.isPresent() && isDatalakeCreatedAfterDataHub(datalakeStack.get(), stack);
+            boolean resizeEntitlementEnabled = entitlementService.isDatalakeLightToMediumMigrationEnabled(ThreadBasedUserCrnProvider.getAccountId());
+            LOGGER.info("Is cluster DH: {}, Found data lake stack: {}, Is DL rebuild: {},  Is resize entitlement Enabled: {}",
+                    clusterDataHub,
+                    datalakeStack.isPresent(),
+                    dlIsRebuild,
+                    resizeEntitlementEnabled);
+            if (clusterDataHub && datalakeStack.isPresent() &&
+                    dlIsRebuild &&
+                    resizeEntitlementEnabled) {
                 LOGGER.info("Triggering update of remote data context");
                 apiConnectors.getConnector(stack).startClusterMgmtServices();
                 clusterBuilderService.configureManagementServices(stack.getId());
-                //Update Hive service configuration
-                Cluster cluster = clusterService.getById(datalakeStack.get().getCluster().getId());
-                Optional<RDSConfig> rdsConfig = postgresConfigService.createRdsConfigIfNeeded(datalakeStack.get(), cluster, DatabaseType.HIVE)
-                        .stream().filter(config -> config.getType().toLowerCase().equals(DatabaseType.HIVE.toString().toLowerCase()))
-                        .findFirst();
-                try {
-                    if (rdsConfig.isEmpty()) {
-                        throw new CloudbreakException("Could not find RDS configuration for Hive");
+                if (apiConnectors.getConnector(stack).doesServiceExist(HIVE_SERVICE)) {
+                    //Update Hive service configuration
+                    Cluster cluster = clusterService.getById(datalakeStack.get().getCluster().getId());
+                    Optional<RDSConfig> rdsConfig = postgresConfigService.createRdsConfigIfNeeded(datalakeStack.get(), cluster, DatabaseType.HIVE)
+                            .stream().filter(config -> config.getType().toLowerCase().equals(DatabaseType.HIVE.toString().toLowerCase()))
+                            .findFirst();
+                    try {
+                        if (rdsConfig.isEmpty()) {
+                            throw new CloudbreakException("Could not find RDS configuration for Hive");
```

Comment:
Maybe would be better if you switch the condition and just log the message if is empty. The stacktrace does not contain any useful information here. like
if (rdsConfig.present())
  //do the logic
else 
  //log the message

Reply:
Addressed your comment. Also made changes not to throw an exception in the failure case. Instead, I'm logging a failure. I don't want to throw an exception till we are sure that the failure that needs the flow to fail.